clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/description/KubernetesSpinnakerKindMap.java

```diff
@@ -60,35 +62,37 @@ public static SpinnakerKind fromString(String name) {
     }
   }
 
-  private final Map<SpinnakerKind, Set<KubernetesKind>> spinnakerToKubernetes = new HashMap<>();
-  private final Map<KubernetesKind, SpinnakerKind> kubernetesToSpinnaker = new HashMap<>();
+  private final ImmutableMap<SpinnakerKind, ImmutableSet<KubernetesKind>> spinnakerToKubernetes;
```

Comment:
If you make this an `ImmutableSetMultimap`, you can make that constructor a whole lot simpler. I don't think anything else needs to change. (A `Multimap` is essentially just a fancy Guava wrapper around `Map<Key, Collection<Value>>` with some nice convenience methods.)

Reply:
That's awesome, I briefly searched for a cleaner way of doing this and didn't find one...I'll make that change now.